factory/src/factory.rs

```diff
@@ -436,13 +441,17 @@ where
         &self,
         buffer: &Buffer<B>,
         offset: u64,
-        content: &[T],
+        content: &T,
         last: Option<BufferState>,
         next: BufferState,
-    ) -> Result<(), UploadError> {
+    ) -> Result<(), UploadError>
+    where
+        T: ?Sized + 'static,
+    {
         assert!(buffer.info().usage.contains(buffer::Usage::TRANSFER_DST));
 
-        let content_size = content.len() as u64 * std::mem::size_of::<T>() as u64;
+        let content_size = std::mem::size_of_val(content) as _;
```

Comment:
Should this be more explicit ?
```suggestion
        let content_size = std::mem::size_of_val(content) as u64;
```